### DISCUSSION TITLE
Failed Build Nyx Mode

### DISCUSSION THREAD

**Discussion by YuriiCrimson** [AFLplusplus/AFLplusplus#1370] [Q&A]
i build Nyx mode and get Error net/eth.c: In function ‘eth_parse_ipv6_hdr’:
net/eth.c:410:15: warning: array subscript ‘struct ip6_ext_hdr_routing[0]’ is partly outside array bounds of ‘struct ip6_ext_hdr[1]’ [-Warray-bounds]
  410 |     if ((rthdr->rtype == 2) &&
      |          ~~~~~^~~~~~~
net/eth.c:487:24: note: while referencing ‘ext_hdr’
  487 |     struct ip6_ext_hdr ext_hdr;
      |                        ^~~~~~~
net/eth.c:411:15: warning: array subscript ‘struct ip6_ext_hdr_routing[0]’ is partly outside array bounds of ‘struct ip6_ext_hdr[1]’ [-Warray-bounds]
  411 |         (rthdr->len == sizeof(struct in6_address) / 8) &&
      |          ~~~~~^~~~~
net/eth.c:487:24: note: while referencing ‘ext_hdr’
  487 |     struct ip6_ext_hdr ext_hdr;
      |                        ^~~~~~~
net/eth.c:412:15: warning: array subscript ‘struct ip6_ext_hdr_routing[0]’ is partly outside array bounds of ‘struct ip6_ext_hdr[1]’ [-Warray-bounds]
  412 |         (rthdr->segleft == 1)) {
      |          ~~~~~^~~~~~~~~
net/eth.c:487:24: note: while referencing ‘ext_hdr’
  487 |     struct ip6_ext_hdr ext_hdr;
      |                        ^~~~~~~
  AS      pc-bios/optionrom/pvh.o
  CC      pc-bios/optionrom/pvh_main.o
cc1: error: ‘-fcf-protection’ is not compatible with this target
make[2]: *** [/mnt/RoxanneWolf/AFLplusplus/nyx_mode/QEMU-Nyx/rules.mak:69: linuxboot_dma.o] Error 1
make[2]: *** Waiting for unfinished jobs....
  BUILD   pc-bios/optionrom/multiboot.img
cc1: error: ‘-fcf-protection’ is not compatible with this target
make[2]: *** [/mnt/RoxanneWolf/AFLplusplus/nyx_mode/QEMU-Nyx/rules.mak:69: pvh_main.o] Error 1
make[1]: *** [Makefile:542: pc-bios/optionrom/all] Error 2
make[1]: *** Waiting for unfinished jobs....
migration/qemu-file.c: In function ‘qemu_get_buffer_in_place’:
migration/qemu-file.c:587:18: warning: ‘src’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  587 |             *buf = src;
      |             ~~~~~^~~~~

**Comment by vanhauser-thc** [comment]
this is qemu that is failing to build for you. the part that is failing has nothing to do with nyx or afl++.
the error tells you that `-fcf-protection` is not allowed as a compile option. maybe remove that globally and try again to compile.
